homeassistant/components/sensor/haveibeenpwned.py

```diff
@@ -0,0 +1,152 @@
+from homeassistant.helpers.entity import Entity
+import homeassistant.helpers.config_validation as cv
+from homeassistant.util import Throttle
+import homeassistant.util.dt as dt_util
+
+_LOGGER = logging.getLogger(__name__)
+
+DATE_STR_FORMAT = "%Y-%m-%d %H:%M:%S"
+USER_AGENT = "Home Assistant HaveIBeenPwned Sensor Component"
+
+SCAN_INTERVAL = 5
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_EMAIL): vol.All(cv.ensure_list, [cv.string]),
+})
+
+
+# pylint: disable=unused-argument
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the RESTful sensor."""
+    emails = config.get(CONF_EMAIL)
+
+    data = HaveIBeenPwnedData(emails)
+    data.update = Throttle(timedelta(seconds=SCAN_INTERVAL-1))(data.update)
+
+    dev = []
+    for email in emails:
+        dev.append(HaveIBeenPwnedSensor(data, email))
+
+    add_devices(dev)
+
+
+class HaveIBeenPwnedSensor(Entity):
+    """Implementation of HaveIBeenPwnedSensor."""
+
+    def __init__(self, data, email):
+        """Initialize the HaveIBeenPwnedSensor sensor."""
+        self._value = None
+        self._data = data
+        self._email = email
+        self._unit_of_measurement = "Hits"
+        self.update()
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return "Breaches {}".format(self._email)
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit the value is expressed in."""
+        return self._unit_of_measurement
+
+    @property
+    def state(self):
+        """Return the state of the device."""
+        if self._value is None:
+            return STATE_UNKNOWN
+        else:
+            return len(self._value)
+
+    @property
+    def state_attributes(self):
+        """Return the atrributes of the sensor."""
+        val = {}
+        if self._value is None:
+            return val
+
+        for idx, value in enumerate(self._value):
+            tmpname = "breach {}".format(idx+1)
+            tmpvalue = "{} {}".format(
+                value["Title"],
+                dt_util.as_local(dt_util.parse_datetime(
+                    value["AddedDate"])).strftime(DATE_STR_FORMAT))
+            val[tmpname] = tmpvalue
+
+        return val
+
+    def update(self):
+        """Get data for (next) email and set value if it's our email."""
+        self._data.update()
+        if self._data.email == self._email:
+            self._value = self._data.data
+
+
+# pylint: disable=too-few-public-methods
+class HaveIBeenPwnedData(object):
+    """Class for handling the data retrieval."""
+
+    def __init__(self, emails):
+        """Initialize the data object."""
+        self._email_count = len(emails)
+        self._current_index = -1
+        self.data = None
+        self.email = None
+        self._emails = emails
+
+    def update(self):
+        """Get the latest data for current email from REST service."""
+        try:
+            self.data = None
```

Comment:
This is such a weird approach. Just keep a dictionary with per email the latest fetched version.